bodyswap.lua

```diff
@@ -149,9 +126,11 @@ function swapAdvUnit(newUnit)
     setNewAdvNemFlags(newNem)
     configureAdvParty(newNem)
     df.global.adventure.player_id = newNem.id
-    activeUnits[newUnitIndex] = oldUnit
-    activeUnits[oldUnitIndex] = newUnit
+    df.global.world.units.adv_unit = newUnit
     oldUnit.idle_area:assign(oldUnit.pos)
+
+    -- +1 is necessary because in adv mode, the window pos is off by 1 from dwarf mode for some reason
```

Comment:
if this is systematically true, we'll need to fix it in Gui.cpp, not here. Could you test further to see if we're also off by 1 in fort mode?

Reply:
as per https://github.com/DFHack/dfhack/pull/4660 it looks like we need to fix this in `revealInDwarfmodeMap` itself. Would you like to submit that fix?